src/utils/inputs-utils.ts

```diff
@@ -0,0 +1,219 @@
+    }
+    return getIconWithSettingName(config, altSettingName);
+  }
+  return icon;
+}
+
+export function assign(config: InputInterfaceConfig, settingNameTarget: InputSettings, keycode: number): boolean {
+  // first, we need to check if this keycode is already used on another settingName
+  const key = getKeyWithKeycode(config, keycode);
+  if (
+    !config.custom
+    || !key
+    || !canIAssignThisKey(config, key)
+    || !canIOverrideThisSetting(config, settingNameTarget)
+  ) {
+    return false;
+  }
+  const previousSettingName = getSettingNameWithKeycode(config, keycode);
+  // if it was already bound, we delete the bind
+  if (previousSettingName !== -1) {
+    const previousKey = getKeyWithSettingName(config, previousSettingName);
+    if (previousKey) {
+      config.custom[previousKey] = -1;
+    }
+  }
+  // then, we need to delete the current key for this settingName
+  const currentKey = getKeyWithSettingName(config, settingNameTarget);
+  if (currentKey) {
+    config.custom[currentKey] = -1;
+  }
+
+  // then, the new key is assigned to the new settingName
+  const newKey = getKeyWithKeycode(config, keycode);
+  if (newKey) {
+    config.custom[newKey] = settingNameTarget;
+  }
+  return true;
+}
+
+export function swap(config: InputInterfaceConfig, settingNameTarget: InputSettings, keycode: number): boolean {
+  // only for gamepad
+  if (config.padType === "keyboard" || isNil(config.custom)) {
+    return false;
+  }
+  const prev_key = getKeyWithSettingName(config, settingNameTarget);
+  const prev_settingName = prev_key ? getSettingNameWithKey(config, prev_key) : -1;
+
+  const new_key = getKeyWithKeycode(config, keycode);
+  const new_settingName = new_key ? getSettingNameWithKey(config, new_key) : -1;
+
+  if (prev_key) {
+    config.custom[prev_key] = new_settingName;
+  }
+  if (new_key) {
+    config.custom[new_key] = prev_settingName;
+  }
+  return true;
+}
+
+/**
+ * Deletes the existing binding of the specified setting name unless it is a locked binding.
+ *
+ * @param config - The configuration object containing custom settings.
+ * @param settingName - The setting name to delete.
+ * @returns `true` if the binding was deleted succcessfully, `false` otherwise.
+ */
+export function deleteBind(config: InputInterfaceConfig, settingName: InputSettings): boolean {
+  const key = getKeyWithSettingName(config, settingName);
+  if (isNil(config.custom) || isNil(key) || !canIDeleteThisSetting(config, settingName)) {
+    return false;
+  }
+  config.custom[key] = -1;
+  return true;
+}
+
+export function canIAssignThisKey(config: InputInterfaceConfig, key: InputKeys): boolean {
+  const settingName = getSettingNameWithKey(config, key);
+  if (config.keysBlacklist?.includes(key) || (settingName !== -1 && config.settingsBlacklist?.includes(settingName))) {
+    return false;
+  }
+  return true;
+}
+
+export function canIOverrideThisSetting(config: InputInterfaceConfig, settingName: InputSettings): boolean {
```

Comment:
```suggestion
export function canOverrideOrDeleteSetting(config: InputInterfaceConfig, settingName: InputSettings): boolean {
```
(Use VSCode refactor tool for this ofc)